sftp-common.c

```diff
@@ -257,3 +271,35 @@ ls_file(const char *name, const struct stat *st, int remote, int si_units)
 	}
 	return xstrdup(buf);
 }
+
+#ifdef IRONCORE
+/**
+ *  Check the directory/file name to see if it ends with the .iron sharing suffix.
+ *
+ *  @param fname Path name to inspect
+ *  @return int Offset of suffix in fname, or -1 if fname doesn't end with suffix
+ */
+int
+iron_extension_offset(const char * name)
```

Comment:
It seems like it'd be simpler and not particularly worse performing to find the end of the string and then check if  `n-IRON_SECURE_FILE_SUFFIX_LEN >= 0` and `name + (n-IRON_SECURE_FILE_SUFFIX_LEN)` was `IRON_SECURE_FILE_SUFFIX`.

:shrug:

I might be out of practice juggling  chainsaws.